documentation/loginAndLogoutFlow.md

```diff
@@ -0,0 +1,51 @@
+## User login and Logout flow
+
+This document describes the flow on a locally installed GPII system when the user logs in and logs out. First an overview of the endpoints and related semantics are given and this will be followed by a more in-depth technical walkthrough of the flow.
+
+### Overview and APIs
+We support 3 different logon related URLs, namely:
+* `/user/:token/logonChange` - change the logon state of the given token (ie. log in or out)
+ ** If no user is logged in, user <mytoken> will be logged in
+ ** If <mytoken> is logged in, <mytoken> will be logged out
+ ** If another user is logged in, nothing happens
+
+* `/user/:token/login` - log the token in
+ ** If no user is logged in, <mytoken> will be logged in
+ ** If another user is logged in, nothing will happen
+
+* `/user/:token/logout` - log the token out
+ ** If no user is logged in, nothing happens
+ ** if user mytoken is logged in, he will be used out
+ ** If another user is logged in, nothing will happen
+
+In general, the the `logonChange` URL should be used. This will take the appropriate action depending on whether the user is logged into the system or not (ie. if the user is currently logged in, he/she will be logged out, else he/she will be logged in).
+
+The reason for preferring the `/logonChange` URL over the `/logout` and `/login` URL, is that we support multiple user listeners and means of logging in and out of the system, so unless the service calling these URLs does a lot of work checking for currently logged in users, eg. a `/logout` call when no user (or some other user) is logged in becomes meaningless. The implementation allows for these kinds of bogus calls to login and logout, simply resulting in an error response to the request.
+
+The reason for continuing to support the specific /login and /logout is that, instead of only supporting the `/logonChange` URL is that they make sense for some user listeners, for example when using the the USB listener, one wouldn't expect to log a user out when inserting the USB, or log a user in when pulling the USB out. So for that, it seems to make sense that it attempts to log in (ie. call `/login`) or log out (ie. call `/logout` when inserting/removing the USB, respectively
+
+
+
+
+### Technical description
+
+The core part of the flow is defined in two files:
+
+* `UserLogonStateChange.js` which contains thehandling of the logon related endpoints are gathered in the flowmanagers  file and kicks off the process. This file contains individual handlers for the `login`, `logout` and `logonChange` endpoints. These handlers all have the `gpii.request.flowManager.userLogonStateChange.stateChangeHandler` grade, which is the component that contains the functionality for the actual logging in and logging off.
```

Comment:
thehandling, also double verb in this sentence - "which contains ... are gathered"